nifi-registry/nifi-registry-core/nifi-registry-web-api/src/main/java/org/apache/nifi/registry/web/security/authentication/jwt/JwtService.java

```diff
@@ -74,14 +79,24 @@ public String getUserIdentityFromToken(final String base64EncodedToken) throws J
             if (StringUtils.isEmpty(jws.getPayload().getIssuer())) {
                 throw new JwtException("No issuer available in token");
             }
-            return jws.getPayload().getSubject();
+
+            return jws;
         } catch (JwtException e) {
-            final String errorMessage = "There was an error validating the JWT";
-            logger.error(errorMessage, e);
-            throw e;
+            throw new JwtException("There was an error validating the JWT", e);
         }
     }
 
+    public String getUserIdentityFromToken(final Jws<Claims> jws) throws JwtException {
+        return jws.getPayload().getSubject();
+    }
+
+    public Set<String> getUserGroupsFromToken(final Jws<Claims> jws) throws JwtException {
+        @SuppressWarnings("unchecked")
+        ArrayList<String> groupsString = jws.getPayload().get(GROUPS_CLAIM, ArrayList.class);
```

Comment:
This should be declared as `List` instead of `ArrayList`:
```suggestion
        final List<String> groupsString = jws.getPayload().get(GROUPS_CLAIM, ArrayList.class);
```

Reply:
It looks like this could return null, so the groups need to be null-checked before being passed to `new HashSet<>()`